src/octree/octree_test.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(test)]
+mod tests {
+    use crate::color::Color;
+    use crate::errors::Result;
+    use crate::generation::build_octree;
+    use crate::math::Isometry3;
+    use crate::octree::{self, BatchIterator, PointCulling, PointLocation, NUM_POINTS_PER_BATCH};
+    use crate::{Point, PointData};
+    use cgmath::{Matrix3, Point3, Quaternion, Vector3, Zero};
+    use collision::Aabb3;
+    use tempdir::TempDir;
+
+    fn build_test_octree(batch_size: usize) -> Box<octree::Octree> {
+        let default_point = Point {
+            position: Vector3::new(-2_699_182.0, -4_294_938.0, 3_853_373.0), //ECEF parking lot porter dr
+            color: Color {
+                red: 255,
+                green: 0,
+                blue: 0,
+                alpha: 255,
+            },
+            intensity: None,
+        };
+
+        let mut points = vec![default_point; 4 * batch_size];
+        points[3 * batch_size].position = Vector3::new(-2_702_846.0, -4_291_151.0, 3_855_012.0); // ECEF STANFORD
+
+        let p = Point3::new(6_400_000.0, 6_400_000.0, 6_400_000.0);
+        let bounding_box = Aabb3::new(-1.0 * p, p);
+
+        let pool = scoped_pool::Pool::new(10);
+        let tmp_dir = TempDir::new("octree").unwrap();
+
+        build_octree(&pool, &tmp_dir, 1.0, bounding_box, points.into_iter());
+        crate::octree::on_disk::octree_from_directory(tmp_dir.into_path()).unwrap()
+    }
+
+    #[test]
+    #[ignore]
```

Comment:
I don't understand why this test is ignored now – does your change impact the batch iterator?

Reply:
the test tooks something longer to be tested. should that always be running?

Reply:
test streamlined